evm/src/executor/inspector/cheatcodes/env.rs

```diff
@@ -104,6 +104,42 @@ fn accesses(state: &mut Cheatcodes, address: Address) -> Bytes {
     }
 }
 
+#[derive(Clone, Debug, Default)]
+pub struct RecordedLogs {
+    pub entries: Vec<RawLog>,
+}
+
+fn start_record_logs(state: &mut Cheatcodes) {
+    state.recorded_logs = Some(Default::default());
+}
+
+fn get_recorded_logs(state: &mut Cheatcodes) -> Bytes {
+    if let Some(recorded_logs) = &mut state.recorded_logs {
+        let tuples = recorded_logs
+            .entries
+            .iter()
+            .map(|entry| {
+                Token::Tuple(
+                    [
+                        Token::Array(
```

Comment:
`Vec<H256>` and `Vec<u8>` implement the `Tokenizable` trait, which also means it implements the `Tokenize` trait, so I think it should be possible to just call `entry.topics.into_tokens()` and `entry.data.into_tokens()`?

Reply:
Some hair pulling here, but managed to refashion this part of the code

```rs
        ethers::abi::encode(
            &recorded_logs
                .entries
                .iter()
                .map(|entry| {
                    Token::Tuple(vec![
                        entry.topics.clone().into_token(),
                        Token::Bytes(entry.data.clone()),
                    ])
                })
                .collect::<Vec<Token>>()
                .into_tokens(),
        )
        .into()
```

Sample output object:

> This maps to `(bytes32[], bytes)[]` or `Log[]` with `struct Log {bytes32[] topics; bytes data;}`

```rs
[ Tuple([
  Array([
    FixedBytes([97, 251, 125, ...]), 
    FixedBytes([... 0, 1])]), 
  Bytes([... 128, 64, 178, 137, 171, 233, 51, 52, 165, 227, ...])]), 
  Tuple([
    Array([
      FixedBytes([122, 249, 45, ...]), 
      FixedBytes([... 0, 2]), 
      FixedBytes([... 0, 3])]), 
    Bytes([... 171, 233, 51, 52, 165, 227, 102, 39, 81, 170, ...])]), 
...
```

* Using both `.into_token()` and `into_tokens()`.
* `into_tokens()` gives you as output a `[Array[ ....`
* `entry.data.clone().into_token()` will give you `Array([Uint(0), Uint(0),` instead of the desired `Bytes([0, 0, 0, 0, 0, `